mlflow/gateway/client.py

```diff
@@ -344,3 +345,77 @@ def query(self, route: str, data: Dict[str, Any]):
                 raise MlflowException(message=timeout_message, error_code=BAD_REQUEST)
             else:
                 raise e
+
+    @experimental
+    def set_limits(self, route: str, limits: List[Dict[str, Any]] = []) -> LimitsConfig:
+        """
+        Set limits on an existing route in the Gateway.
+
+        .. warning::
+
+            This API is **only available** when running within Databricks.
+
+        :param route_name: The name of the route to set limits on.
+        :param limits: Limits to set on the route.
+
+        :raises mlflow.MlflowException: If the function is not running within Databricks.
+
+        Example usage from within Databricks:
+
+        .. code-block:: python
+
+            from mlflow.gateway import set_gateway_uri, set_limits
+
+            set_gateway_uri(gateway_uri="databricks")
+
+            set_limits("my-new-route", [{"key": "user", "renewal_period": "minute", "calls": 50}])
+
+        """
+
+        if not self._is_databricks_host():
+            raise MlflowException(
+                "The set_limits API is only available when running within Databricks.",
```

Comment:
This error message sounds misleading because we can run this API on your laptop against Databricks using a token.

Reply:
changed to "when runnning against Databricks AI Gateway".